grid2op/tests/test_alert_trust_score.py

```diff
@@ -0,0 +1,1767 @@
+from grid2op.Parameters import Parameters
+from grid2op.Exceptions import Grid2OpException
+from grid2op.Runner import Runner  # TODO
+from grid2op.Opponent import BaseOpponent, GeometricOpponent, GeometricOpponentMultiArea
+from grid2op.Action import BaseAction, PlayableAction
+from grid2op.Agent import BaseAgent
+from grid2op.Episode import EpisodeData
+
+
+ALL_ATTACKABLE_LINES= [
+            "62_58_180",
+            "62_63_160",
+            "48_50_136",
+            "48_53_141",
+            "41_48_131",
+            "39_41_121",
+            "43_44_125",
+            "44_45_126",
+            "34_35_110",
+            "54_58_154",
+        ] 
+
+ATTACKED_LINE = "48_50_136"
+
+
+#DEFAULT_PARAMS_TRUSTSCORE = dict(reward_min_no_blackout=-1.0,
+#                                 reward_min_blackout=-10.0,
+#                                 reward_max_no_blackout=1.0,
+#                                 reward_max_blackout=2.0,
+#                                 reward_end_episode_bonus=42.0,
+#                                 min_score=-1.0)
+
+DEFAULT_PARAMS_TRUSTSCORE = dict(reward_min_no_blackout=-1.0,
+                                 reward_min_blackout=-50.0,
+                                 reward_max_no_blackout=0.0,
+                                 reward_max_blackout=0.0,
+                                 reward_end_episode_bonus=0.0,
+                                 min_score=-3.0)
+
+def _get_steps_attack(kwargs_opponent, multi=False):
+    """computes the steps for which there will be attacks"""
+    ts_attack = np.array(kwargs_opponent["steps_attack"])
+    res = []
+    for i, ts in enumerate(ts_attack):
+        if not multi:
+            res.append(ts + np.arange(kwargs_opponent["duration"]))
+        else:
+            res.append(ts + np.arange(kwargs_opponent["duration"][i]))
+    return np.unique(np.concatenate(res).flatten())
+
+    
+class TestOpponent(GeometricOpponent): 
+    """An opponent that can select the line attack, the time and duration of the attack."""
+    
+    def __init__(self, action_space):
+        super().__init__(action_space)
+        self.custom_attack = None
+        self.duration = None
+        self.steps_attack = None
+
+    def init(self, partial_env,  lines_attacked=[ATTACKED_LINE], duration=10, steps_attack=[0,1]):
+        super().init(partial_env, lines_attacked, )
+        attacked_line = lines_attacked[0]
+        self.custom_attack = self.action_space({"set_line_status" : [(l, -1) for l in lines_attacked]})
+        self.duration = duration
+        self.steps_attack = steps_attack
+        self.env = partial_env
+
+    def attack(self, observation, agent_action, env_action, budget, previous_fails): 
+        if observation is None:
+            return None, None
+        current_step = self.env.nb_time_step
+        if current_step not in self.steps_attack: 
+            return None, None
+        
+        return self.custom_attack, self.duration
+
```

Comment:
no copy paste

Reply:
It is apparently an attackant sending custom attacks.
What should be done in that case ?